modules/porous_flow/test/tests/gravity/grav02e_fv.i

```diff
@@ -0,0 +1,226 @@
+    fv = true
+  []
+[]
+
+[FVKernels]
+  [mass0]
+    type = FVPorousFlowMassTimeDerivative
+    fluid_component = 0
+    variable = ppwater
+  []
+  [flux0]
+    type = FVPorousFlowAdvectiveFlux
+    fluid_component = 0
+    variable = ppwater
+  []
+  [mass1]
+    type = FVPorousFlowMassTimeDerivative
+    fluid_component = 1
+    variable = sgas
+  []
+  [flux1]
+    type = FVPorousFlowAdvectiveFlux
+    fluid_component = 1
+    variable = sgas
+  []
+[]
+
+[AuxKernels]
+  [ppgas]
+    type = ADPorousFlowPropertyAux
+    property = pressure
+    phase = 1
+    variable = ppgas
+    execute_on = 'initial timestep_end'
+  []
+  [swater]
+    type = ADPorousFlowPropertyAux
+    property = saturation
+    phase = 0
+    variable = swater
+    execute_on = 'initial timestep_end'
+  []
+  [relpermwater]
+    type = ADPorousFlowPropertyAux
+    property = relperm
+    phase = 0
+    variable = relpermwater
+    execute_on = 'initial timestep_end'
+  []
+  [relpermgas]
+    type = ADPorousFlowPropertyAux
+    property = relperm
+    phase = 1
+    variable = relpermgas
+    execute_on = 'initial timestep_end'
+  []
+[]
+
+[UserObjects]
+  [dictator]
+    type = PorousFlowDictator
+    porous_flow_vars = 'ppwater sgas'
+    number_fluid_phases = 2
+    number_fluid_components = 2
+  []
+  [pc]
+    type = PorousFlowCapillaryPressureConst
+    pc = 1e5
+  []
+[]
+
+[FluidProperties]
+  [simple_fluid0]
+    type = SimpleFluidProperties
+    bulk_modulus = 2e9
+    density0 = 1000
+    viscosity = 1e-3
+    thermal_expansion = 0
+  []
+  [simple_fluid1]
+    type = SimpleFluidProperties
+    bulk_modulus = 2e9
+    density0 = 10
+    viscosity = 1e-5
+    thermal_expansion = 0
+  []
+[]
+
+[Materials]
+  [temperature]
+    type = ADPorousFlowTemperature
+  []
+  [ppss]
+    type = ADPorousFlow2PhasePS
+    phase0_porepressure = ppwater
+    phase1_saturation = sgas
+    capillary_pressure = pc
+  []
+  [massfrac]
+    type = ADPorousFlowMassFraction
+    mass_fraction_vars = 'massfrac_ph0_sp0 massfrac_ph1_sp0'
+  []
+  [simple_fluid0]
+    type = ADPorousFlowSingleComponentFluid
+    fp = simple_fluid0
+    phase = 0
+  []
+  [simple_fluid1]
+    type = ADPorousFlowSingleComponentFluid
+    fp = simple_fluid1
+    phase = 1
+  []
+  [porosity]
+    type = ADPorousFlowPorosityConst
+    porosity = 0.1
+  []
+  [permeability]
+    type = ADPorousFlowPermeabilityConst
+    permeability = '1e-11 0 0 0 1e-11 0  0 0 1e-11'
+  []
+  [relperm_water]
+    type = ADPorousFlowRelativePermeabilityCorey
+    n = 2
+    phase = 0
+  []
+  [relperm_gas]
+    type = ADPorousFlowRelativePermeabilityCorey
+    n = 2
+    phase = 1
+  []
+[]
+
+[VectorPostprocessors]
+  [vars]
+    type = ElementValueSampler
+    variable = 'ppgas ppwater sgas swater'
+    sort_by = x
+  []
+[]
+
+[Preconditioning]
+  [smp]
+    type = SMP
+    full = true
+    petsc_options_iname = '-ksp_type -pc_type -snes_atol -snes_rtol'
+    petsc_options_value = 'bcgs bjacobi 1E-12 1E-10'
```

Comment:
bcgs, fancy!

Reply:
how do you choose when to do `bcgs` vs `gmres` ?

Reply:
Haha - copy and paste from an existing test! It probably doesn't even need any if these options to work - I'll see if I can remover them

Reply:
I think it is ok to just delete these options